packages/toolpad-utils/src/collections.ts

```diff
@@ -93,3 +95,66 @@ export function filterKeys<U>(
 ): Record<string, U> {
   return mapProperties(obj, ([key, value]) => (filter(key) ? [key, value] : null));
 }
+
+/**
+ * Check if two values are equal - that is,
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isReferEqual(obj: any, objToCompare: any): boolean {
```

Comment:
I would maybe call the function `isDeepEqual` since we're checking for deep equality here, not referential equality.
https://www.syncfusion.com/blogs/post/5-different-ways-to-deep-compare-javascript-objects.aspx